docs/applications.md

```diff
@@ -203,3 +203,42 @@ within FCM's limit.
 Please also note that some fields will be unavailable if you registered a pusher
 with `event_id_only` format.
 
+### Web Push
+
+#### Setup & configuration
+
+In the sygnal virtualenv, generate the server key pair by running `vapid --gen --applicationServerKey`. This will generate a `private_key.pem` (which you'll refer to in the config file with `vapid_private_key`) and `public_key.pem` file, and also string labeled `Application Server Key`. You'll copy the Application Server Key to your web application to subscribe to the push manager:
+
+```js
+serviceWorkerRegistration.pushManager.subscribe({
+    userVisibleOnly: true,
+    applicationServerKey: "...",
+});
+```
+
+You also need to set an e-mail address in `vapid_contact_email` in the config file, where the push gateway operator can reach you in case they need to notify you about your usage of their API.
+
+#### Push key and expected push data
+
+In your web application, [the push manager subscribe method](https://developer.mozilla.org/en-US/docs/Web/API/PushManager/subscribe) will return [a subscription](https://developer.mozilla.org/en-US/docs/Web/API/PushSubscription) with an `endpoint` and `keys` property, the latter containing a `p256dh` and `auth` property. The `p256dh` key is used as the push key, and the push data is expected `endpoint` and `auth`. You can also set `default_payload` in the push data; any properties set in it will be present in the push messages you receive, so it can be used to pass identifiers specific to your client (like which account the notification is for).
+
+Also note that because you can only have one push subscription per service worker, and hence per origin, you might create pushers for different accounts with the same p256dh push key. To prevent the server from removing other pushers with the same push key for your other users, you should set `append` to `true` when uploading your pusher.
+
+#### Notification format
+
+The notification as received by your web application will contain these keys if they were set by the homeserver, and otherwise omit them.
```

Comment:
This feels like it should link to the spec somewhere instead of listing them out again?

Reply:
I've added them here again since every pushkin seems to have their own format they push to their clients, and the format is slightly different than the spec (counts is flattened). I've mentioned this in the docs now and linked to the spec.